File: app/src/main/java/com/example/androiddrawingpad/DrawingActivity.java
import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;

public class DrawingActivity extends AppCompatActivity {

    private DrawingView drawingView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_drawing);

        drawingView = findViewById(R.id.drawingView);

        // Set up color buttons
        findViewById(R.id.colorRed).setOnClickListener(v -> drawingView.setColor(Color.RED));
        findViewById(R.id.colorGreen).setOnClickListener(v -> drawingView.setColor(Color.GREEN));
        findViewById(R.id.colorBlue).setOnClickListener(v -> drawingView.setColor(Color.BLUE));

        // Set up width buttons
        findViewById(R.id.widthSmall).setOnClickListener(v -> drawingView.setStrokeWidth(5f));
        findViewById(R.id.widthMedium).setOnClickListener(v -> drawingView.setStrokeWidth(10f));
        findViewById(R.id.widthLarge).setOnClickListener(v -> drawingView.setStrokeWidth(20f));

        // Set up clear button
        findViewById(R.id.clearButton).setOnClickListener(v -> drawingView.clear());
    }
}

File: app/src/main/java/com/example/androiddrawingpad/DrawingActivity.kt
import android.graphics.Color
import android.os.Bundle
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import com.example.androiddrawingpad.R
import android.widget.Button
import android.widget.ImageButton



class DrawingActivity : AppCompatActivity() {

    private lateinit var drawingView: DrawingView
    private lateinit var colorRed: ImageButton
    private lateinit var colorGreen: ImageButton
    private lateinit var colorBlue: ImageButton
    private lateinit var widthSmall: Button
    private lateinit var widthMedium: Button
    private lateinit var widthLarge: Button
    private lateinit var clearButton: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_drawing)


        val startButton: Button = findViewById(R.id.startButton)
        startButton.setOnClickListener {
            val intent = Intent(this, DrawingActivity::class.java)
            startActivity(intent)
        }

        drawingView = findViewById(R.id.drawingView)
        colorRed = findViewById(R.id.colorRed)
        colorGreen = findViewById(R.id.colorGreen)
        colorBlue = findViewById(R.id.colorBlue)
        widthSmall = findViewById(R.id.widthSmall)
        widthMedium = findViewById(R.id.widthMedium)
        widthLarge = findViewById(R.id.widthLarge)
        clearButton = findViewById(R.id.clearButton)

        
        colorRed.setOnClickListener { drawingView.setColor(Color.RED) }
        colorGreen.setOnClickListener { drawingView.setColor(Color.GREEN) }
        colorBlue.setOnClickListener { drawingView.setColor(Color.BLUE) }

        
        widthSmall.setOnClickListener { drawingView.setStrokeWidth(5f) }
        widthMedium.setOnClickListener { drawingView.setStrokeWidth(10f) }
        widthLarge.setOnClickListener { drawingView.setStrokeWidth(20f) }

       
        clearButton.setOnClickListener { drawingView.clear() }
    }
}
